plugins/modules/net_tools/dnsimple_info.py

```diff
@@ -252,84 +250,85 @@ def build_url(account, key, is_sandbox):
-        data = response.json()["data"]
-        pages = response.json()["pagination"]["total_pages"]
-        if int(pages) > 1:
-            for page in range(1, pages):
-                page = page + 1
-                request_object.url = base_url + '&page=' + str(page)
-                new_results = Session().send(request_object)
-                data = data + new_results.json()["data"]
-        return(data)
-    else:
+    if 'pagination' not in response.json():
         module.fail_json('API Call failed, check ID, key and sandbox values')
 
+    data = response.json()['data']
+    total_pages = int(response.json()['pagination']['total_pages'])
+    page = 1
+    while page < total_pages:
+        page = page + 1
+        request_object.url = '{url}&page={page}'.format(url=base_url, page=page)
+        new_results = Session().send(request_object)
+        data = data + new_results.json()['data']
+
+    return data
+
 
 def record_info(dnsimple_mod, req_obj):
-    req_obj.url, req_obj.method = req_obj.url + '/zones/' + dnsimple_mod.params["name"] + '/records?name=' + dnsimple_mod.params["record"], 'GET'
+    req_obj.url, req_obj.method = '{url}/zones/{name}/records/?name={record}'.format(
+        url=req_obj.url,
+        name=dnsimple_mod.params['name'],
+        record=dnsimple_mod.params['record']), 'GET'
     return iterate_data(dnsimple_mod, req_obj)
 
 
 def domain_info(dnsimple_mod, req_obj):
-    req_obj.url, req_obj.method = req_obj.url + '/zones/' + dnsimple_mod.params["name"] + '/records?per_page=100', 'GET'
+    req_obj.url, req_obj.method = '{url}/zones/{name}/records?per_page=100'.format(url=req_obj.url, name=dnsimple_mod.params['name']), 'GET'
     return iterate_data(dnsimple_mod, req_obj)
 
 
 def account_info(dnsimple_mod, req_obj):
-    req_obj.url, req_obj.method = req_obj.url + '/zones/?per_page=100', 'GET'
+    req_obj.url, req_obj.method = '{url}/zones/?per_page=100'.format(url=req_obj.url), 'GET'
     return iterate_data(dnsimple_mod, req_obj)
 
 
 def main():
-    # define available arguments/parameters a user can pass to the module
-    fields = {
-        "account_id": {"required": True, "type": "str"},
-        "api_key": {"required": True, "type": "str", "no_log": True},
-        "name": {"required": False, "type": "str"},
-        "record": {"required": False, "type": "str"},
-        "sandbox": {"required": False, "type": "bool", "default": False}
-    }
-
     result = {
         'changed': False
     }
 
     module = AnsibleModule(
-        argument_spec=fields,
-        supports_check_mode=True
+        argument_spec={
+            'account_id': {'required': True, 'type': 'str'},
+            'api_key': {'required': True, 'type': 'str', 'no_log': True},
+            'name': {'type': 'str'},
+            'record': {'type': 'str'},
+            'sandbox': {'type': 'bool', 'default': False}
+        },
+        supports_check_mode=True,
+        required_by={
+            'record': 'name',
+        }
```

Comment:
How about:
```suggestion
        }
        required_one_of=[
            ('api_key', 'account_id'),
        ],
```

Reply:
I thought of that, but I thought you would pick on this being not backwards compatible: `required_one_of` tests on the basis of `None`-ness, whereas the code tests for `True`-ness. Hence if some user passes `account_id: ""`, the code currently fails immediately, whereas with `required_one_of` it would place the request to the API and then fail.

As I write, I think we might add both.

Reply:
Sorry, I had forgotten: both parameters are `required`, so writing a `required_one_of` is not only semantically incorrect, but also redundant. Rolling back.

Reply:
Ah, ok. I also missed the 'required'. Let's keep it as-is then.